test/prompt/test_prompt_node.py

```diff
@@ -250,6 +250,23 @@ def test_generation_kwargs_from_prompt_node_call():
         )
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize("prompt_model", ["google/flan-t5-base", "gpt-3.5-turbo"], indirect=True)
```

Comment:
@Faaany Let's check node.prompt_model.model_invocation_layer.pipe for HF only. For other models we can check generic invoke. Something like:

```
with patch.object(node.prompt_model.model_invocation_layer, "invoke", MagicMock())
...
...
```

Reply:
How about this?

```python
@pytest.mark.integration
@pytest.mark.parametrize("prompt_model", ["hf", "openai", "azure"], indirect=True)
def test_generation_kwargs_from_prompt_node_run_hf(prompt_model):
    skip_test_for_invalid_key(prompt_model)
    the_question = "What does 42 mean?"
    # test that generation_kwargs are passed to the underlying invocation layer
    node = PromptNode(prompt_model)
    with patch.object(node.prompt_model.model_invocation_layer, "invoke", MagicMock()) as mock_call:
        node.run(query=the_question, prompt_template="{query}", generation_kwargs={"do_sample": True, "test": True})

        mock_call.assert_called_with(
            prompt=the_question,
            stop_words=None,
            top_k=1,
            query=the_question,
            generation_kwargs={"do_sample": True, "test": True}
        )
```

Reply:
Yes, great @Faaany  that's exactly what I had on mind as well